pkg/scalers/newrelic_scaler.go

```diff
@@ -0,0 +1,181 @@
+}
+
+type newrelicMetadata struct {
+	account     int
+	region      string
+	queryKey    string
+	metricName  string
+	nrql        string
+	threshold   int
+	scalerIndex int
+}
+
+var newrelicLog = logf.Log.WithName("new-relic_scaler")
+
+func NewNewRelicScaler(config *ScalerConfig) (Scaler, error) {
+	meta, err := parseNewRelicMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing new-relic metadata: %s", err)
+	}
+
+	nrClient, err := newrelic.New(
+		newrelic.ConfigPersonalAPIKey(meta.queryKey),
+		newrelic.ConfigRegion(meta.region))
+
+	if err != nil {
+		log.Fatal("error initializing client:", err)
+	}
+
+	logMsg := fmt.Sprintf("Initializing New Relic Scaler (account %d in region %s)", meta.account, meta.region)
+
+	newrelicLog.Info(logMsg)
+
+	return &newrelicScaler{
+		metadata: meta,
+		nrClient: nrClient}, nil
+}
+
+func parseNewRelicMetadata(config *ScalerConfig) (*newrelicMetadata, error) {
+	meta := newrelicMetadata{}
+	var err error
+	if val, ok := config.TriggerMetadata[account]; ok && val != "" {
+		t, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %s: %s", account, err)
+		}
+		meta.account = t
+	} else {
+		return nil, fmt.Errorf("no %s given >%s<", account, val)
+	}
+
+	meta.queryKey, err = GetFromAuthOrMeta(config, queryKey)
+	if err != nil {
+		return nil, fmt.Errorf("no %s given", queryKey)
+	}
+
+	if val, ok := config.TriggerMetadata[metricName]; ok && val != "" {
+		meta.metricName = val
+	} else {
+		return nil, fmt.Errorf("no %s given", metricName)
+	}
+
+	if val, ok := config.TriggerMetadata[nrql]; ok && val != "" {
+		meta.nrql = val
+	} else {
+		return nil, fmt.Errorf("no %s given", nrql)
+	}
+
+	meta.region, err = GetFromAuthOrMeta(config, region)
+	if err != nil {
+		meta.region = "US"
+		newrelicLog.Info("Using default \"US\" region")
+	}
+
+	if val, ok := config.TriggerMetadata[threshold]; ok && val != "" {
+		t, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %s: %s", threshold, err)
+		}
+		meta.threshold = t
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+	return &meta, nil
+}
+
+func (s *newrelicScaler) IsActive(ctx context.Context) (bool, error) {
+	val, err := s.ExecuteNewRelicQuery(ctx)
+	if err != nil {
+		newrelicLog.Error(err, "error executing newrelic query")
+		return false, err
+	}
+	return val > 0, nil
+}
+
+func (s *newrelicScaler) Close(context.Context) error {
+	return nil
+}
+
+func (s *newrelicScaler) ExecuteNewRelicQuery(ctx context.Context) (float64, error) {
+	nrdbQuery := nrdb.NRQL(s.metadata.nrql)
+	resp, err := s.nrClient.Nrdb.QueryWithContext(ctx, s.metadata.account, nrdbQuery)
+	if err != nil {
+		newrelicLog.Error(err, "error running NerdGraph query")
+		return 0, fmt.Errorf("query return no results")
+	}
+	for _, r := range resp.Results {
+		val, ok := r[s.metadata.metricName].(float64)
+		if ok {
+			newrelicLog.Info("Result of the query %s is %s", s.metadata.nrql, val)
```

Comment:
This logs seems like a debug message, am I right?

Reply:
Totally right, same reply as above comment.

Reply:
no rush at all :)